controllers/utils/affinity_group.go

```diff
@@ -88,12 +89,19 @@ func (r *ReconciliationRunner) GetOrCreateAffinityGroup(
 }
 
 // The computed affinity group name relevant to this machine.
-func GenerateAffinityGroupName(csm infrav1.CloudStackMachine, capiMachine *clusterv1.Machine) (string, error) {
+func GenerateAffinityGroupName(csm infrav1.CloudStackMachine, capiMachine *clusterv1.Machine, capiCluster *clusterv1.Cluster) (string, error) {
 	managerOwnerRef := GetManagementOwnerRef(capiMachine)
 	if managerOwnerRef == nil {
 		return "", errors.Errorf("could not find owner UID for %s/%s", csm.Namespace, csm.Name)
 	}
 	titleCaser := cases.Title(language.English)
-	return fmt.Sprintf("%sAffinity-%s-%s-%s",
-		titleCaser.String(csm.Spec.Affinity), managerOwnerRef.Name, managerOwnerRef.UID, csm.Spec.FailureDomainName), nil
+
+	// If the machine's owner is KubeadmControlPlane or EtcdadmCluster, then we don't consider the name and UID of the
+	// owner, since there will only be one of each of those per cluster.
+	if managerOwnerRef.Kind == "KubeadmControlPlane" || managerOwnerRef.Kind == "EtcdadmCluster" {
+		return fmt.Sprintf("%s-%s-%sAffinity-%s-%s",
+			capiCluster.Name, capiCluster.UID, titleCaser.String(csm.Spec.Affinity), managerOwnerRef.Kind, csm.Spec.FailureDomainName), nil
+	}
+	return fmt.Sprintf("%s-%s-%sAffinity-%s-%s-%s",
```

Comment:
@vishesh92 
with the new format, might it cause backwards compatibility issue ?

Reply:
There is no support for updating the affinity group as of now. Existing machines will continue working with their old affinity group.
Newer machines in that machineset will have the different affinity group from machines before upgrade of CAPC.

Can we go with this approach and add a message in release notes regarding this?